### ISSUE TITLE
Pillarized schedule not removed when missing

### ISSUE THREAD

**Issue by syphernl** [saltstack/salt#21337]
We have a schedule in our pillar which is applied to `*` telling minions to highstate every 15 minutes and return the data to a returner.

After disabling this for a few minions (by adding a grain check to the `schedule.sls`) they were still being highstated every 15 minutes. 

As a test I have restarted the minion on one of the excluded machines and it properly stopped highstating while the others were still doing that.

For another minion I have tried to forcefully refresh the pillars by issuing `saltutil.refresh_pillar` which did resolve it for that. Could this be some kind of cache in the minion? Generally pillar changes are picked up automatically without needing a manual refresh, but perhaps the actual refresh does something specific when it contains a schedule?

**Comment by garethgreenaway** [comment]
What version of Salt?  Can you provide the output from salt --versions-report and what the schedule.sls looks like?

**Comment by syphernl** [comment]
This was on 2014.7.0:
```
           Salt: 2014.7.0
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.3.0
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.0.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: 0.9.1
```

The schedule is nothing fancy:
```
{%- if salt['grains.get']('salt:scheduler', True) == True %}
schedule:
  highstate:
    function: state.highstate
    returner: dashboard
    minutes: 15
    maxrunning: 1
{%- endif %}
```

**Comment by jfindlay** [comment]
@syphernl, have you tried 2014.7.2?

**Comment by syphernl** [comment]
No, I cannot until it is available in the PPA unfortunately.

**Comment by jfindlay** [comment]
:-(

**Comment by jfindlay** [comment]
@syphernl, this might have been fixed by 2014.7.2.  If you have a chance to try out that version, update here when you do, thanks.
